hibernate-core/src/main/java/org/hibernate/query/sqm/sql/internal/StandardSqmSelectTranslator.java

```diff
@@ -271,6 +273,7 @@ public ModelPart resolveModelPart(NavigablePath navigablePath) {
 	@Override
 	public List<Fetch> visitFetches(FetchParent fetchParent) {
 		final List<Fetch> fetches = CollectionHelper.arrayList( fetchParent.getReferencedMappingType().getNumberOfFetchables() );
+		final List<String> bagRoles = new ArrayList<>();
```

Comment:
It might be considerable to specify the `bagRoles`'s initial capacity to 2 to save some memory (2 should be enough for most cases). Super minor though.